### PR TITLE
allows the datepicker to determine the validity of a date by props.minDate

### PR DESCRIPTION
**Pull request by zxiang77**
This PR is my proposed solution to address what I mentioned in the issue I just created a while back.  https://github.com/Hacker0x01/react-datepicker/issues/2834

**Impact:**
It will allows the calendar to be auto-updated if the props.minDate is set, and the year is greater than the year in props.minDate even the year is smaller than 1000

**Tests:**
WIP
will check CI and see if my PR breaks anything

### CODE REVIEW

**Comment by martijnrusschen** [comment]
Can you add some tests to cover the functionality?

**Comment by zxiang77** [comment]
> Can you add some tests to cover the functionality?

sure! I'll try adding it over the weekend. thanks for the reminder

File: test/date_utils_test.js [hunk 1]
```diff
@@ -868,7 +868,7 @@ describe("date_utils", function() {
     });
 
     it("should return the 4 2021 year week", () => {
-      const date = new Date("2021-01-18");
+      const date = new Date("2021-01-19");
```
Comment (zxiang77) [inline test/date_utils_test.js:871]: the test was failing likely due to locale issue. had to increment the date by 1 to make it pass

**Comment by martijnrusschen** [comment]
The test suite is still failing, please try to rebase with master I think that should solve it.

**Comment by zxiang77** [comment]
@martijnrusschen merged master and all the tests are passing now fyi. thanks for ur help!

**Review by martijnrusschen — APPROVED** [review]
Sorry for the delay, can fix the merge conflict so I can get this merged?

**Pull request merged**